compiler/rustc_metadata/src/native_libs.rs

```diff
@@ -196,6 +197,15 @@ impl ItemLikeVisitor<'tcx> for Collector<'tcx> {
                 .span_label(m.span, "missing `name` argument")
                 .emit();
             }
+
+            if lib.kind == NativeLibKind::RawDylib {
```

Comment:
It'd be great to add a diagnostic note or error here if the target is 32 bit windows and `Abi != C && Abi != cdecl`. Beats a confusing linker message later down the line.